structured_prediction_baselines/modules/loss/nce_loss.py

```diff
@@ -104,10 +105,14 @@ def compute_loss(
         distance = self.distance(
             y, y_hat.expand_as(y)
         )  # (batch, 1+num_samples) # does the job of Pn
-        score = self.score_nn(
-            x, y, buffer
-        )  # type:ignore # (batch, 1+num_samples)
-        assert not distance.requires_grad
+        if self.use_scorenn:
+            score = self.score_nn(
+                x, y, buffer
+            )  # type:ignore # (batch, 1+num_samples)
+            assert not distance.requires_grad
+        else:
+            score = 0 
+                    
```

Comment:
Why set the score to 0 in the else case? Shouldn't the the `score = self.score_nn(..)` be before the if-else because that should be executed always.

Reply:
To directly apply ranking loss to the taskNN, I wanted to share the NCE loss classes.

To do that, for the score- distance 
I have to now have - distance = probability,  and score to be 0. 

To sum up, in the config,  I am setting the loss as follows.
```
          {
            type: 'multi-label-nce-ranking-with-discrete-sampling',
            log_key: 'nce-on-tasknn',
            num_samples: num_samples,
            sign: '+', // score + sample prob --> sample prob (as score=0 in this setting.)
            use_scorenn: false, // score=0 because of use_scorenn = False.
            normalize_y: true,
          },

```

I actually didn't want it to execute it in the case of `self.use_scorenn = False` because I want to just set them to 0.
Would there be any cascaded error that is not directly seen in this code? (FYI, it ran fine and seemed like contributing to better performance.)